packages/docs/community/release-process/smart-contracts.md

```diff
@@ -5,93 +5,110 @@ This release process is a work in progress. Many infrastructure components requi
+
+### Core Contracts
 
-Changes to core smart contracts are made via on-chain Governance, approximately four times a year. When a release is made, **all** smart contracts from the release branch that differ from the deployed smart contracts are released, and included in the **same** governance proposal.
+Every deployed Celo core contract has its current version number as a constant which is publicly accessible via the `getVersion()` function, which returns the storage, major, minor, and patch version. Version number is encoded in the Solidity source and updated as part of code changes.
+
+Celo Core Contracts deployed to a live network without the `getVersion()` function, such as the original set of core contracts, are to be considered version `1.1.0.0`.
 
 ### Mixins and libraries
+
 Mixin contracts and libraries are considered part of the contracts that consume them. When a mixin or library has changed, all contracts that consume them should be considered to have changed as well, and thus the contracts should have their version numbers incremented and should be re-deployed as part of the next smart contract release.
 
-## Identifying releases
 
-Each release is identified by a unique monotonically increasing number `N`, with `1` being the first release.
+### Release management in Git/Github
 
-### Git branches
-Every smart contract release has a designated branch, e.g. `release/contracts/N` in the celo-monorepo.
+Github branches/tags and Github releases are used to coordinate past and ongoing releases. Ongoing smart contract development is done on the `master` branch (even after release branches are cut).
 
-Ongoing smart contract development is done on the `master` branch.
+Every smart contract release has a designated release branch, e.g. `release/celo-core-contracts/${N}` in the celo-monorepo. When a new release branch is cut, the commit is tagged with `celo-core-contracts-v${N}.pre-audit`. On Github, a pre-release Github release should be created pointing at the latest tag on the release branch. Ongoing audit responses/fixes should continue to go into `release/celo-core-contracts/${N}`. Whenever release candidates are available they should be tagged `celo-core-contracts-v${N}.rc1`, `...rc2`, etc.
 
-### Github tags
-All release branches should be tagged as such, e.g. `celo-contracts-N`. Each should include a summary of the release contents.
+When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit that was used (which should be the latest release candidate) should be tagged with `celo-core-contracts-v${N}.(baklava | alfajores | mainnet)`.
```

Comment:
If we're deploying the same upgrade, why do we need a branch for each network?

Reply:
So these wouldn't be branches, but tags. As you point out, they would likely point to the same commit. This would just be a way to people to look up which environment has which commit deployed.